### PR TITLE
Explicitly split up the cases when we refresh the authorization credential

### PR DESCRIPTION
**Pull request by seehamrun**
So that we can pin down why we're still getting 401 errors in the worker after the job reaches 60minutes

### CODE REVIEW

File: extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java [hunk 1]
```diff
@@ -180,12 +180,19 @@ BatchMediaItemResponse createPhoto(NewMediaItemUpload newMediaItemUpload) throws
...
+        monitor.info(() -> "Attempting to refresh authorization token");
+        if (credential.refreshToken()) {
+          // if the second attempt throws an error, then something else is wrong, and we bubble up the response errors
+          monitor.info(() -> "Refreshed authorization token successfuly");
+          response = requestFactory
+              .buildPostRequest(new GenericUrl(url + "?" + generateParamsString(parameters)),
+                  httpContent).execute();
+          
+        } else {
+          throw new IOException("Couldn't refresh authorization token!");
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java]: Couldn't refresh authorization token upon retry!

File: extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java [hunk 2]
```diff
@@ -180,12 +180,19 @@ BatchMediaItemResponse createPhoto(NewMediaItemUpload newMediaItemUpload) throws
...
-      // if the response is "unauthorized" and we've successfully refreshed the token, try the request again
-      if (e.getStatusCode() == 401 && credential.refreshToken()) {
-        // if the second attempt throws an error, then something else is wrong, and we bubble up the response errors
-        response = requestFactory
-            .buildPostRequest(new GenericUrl(url + "?" + generateParamsString(parameters)),
-                httpContent).execute();
+      // if the response is "unauthorized", refresh the token and try the request again
+      if (e.getStatusCode() == 401) {
+        monitor.info(() -> "Attempting to refresh authorization token");
+        if (credential.refreshToken()) {
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java]: Do we need a pause here?

Comment (seehamrun) [reply extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java]: I dont think so - refreshToken doesn't return until the token's been set if im reading this correctly: 
https://googleapis.dev/java/google-oauth-client/1.25.0/index.html (ctrl-f for "refreshtoken()")

File: extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java (hunk 1 above)
Comment (seehamrun) [reply extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java]: Done

**Comment by seehamrun** [comment]
Ok finally got travis building - ptal

**Pull request merged**